department-of-veterans-affairs/caseflow
# Add receipt date validation to grant docket switch form

Resolves [CASEFLOW-1055](https://vajira.max.gov/browse/CASEFLOW-1055)

### Description
Add validation for receipt date to Grant Docket Switch Form

### Acceptance Criteria
- [ ] Inline validation error appears if receipt date is before prior-to-RAMP date
- [ ] Inline validation error appears if receipt date is in the future

### Testing Plan
1. Switch to COB user
2. Add mail task for docket switch to an appeal.
*Select grant all issues as recommendation and assign to judge
3. Switch to said Judge 
4. Grant all issues
5. Switch back to COB user and search for the appeal
6. Select the mail task to "Granted Docket Switch"
7. Select receipt date before 11/1/2017
8. Verify inline error appears
9. Select receipt date in the future
10. Verify inline error concerning future date appears


### User Facing Changes

 BEFORE|AFTER
 ---|---
<> | <img width="1105" alt="Screen Shot 2021-03-03 at 4 49 52 PM" src="https://user-images.githubusercontent.com/68879427/109909099-5a5edf80-7c5a-11eb-94a3-6141f4cf7d51.png"><img width="934" alt="Screen Shot 2021-03-03 at 7 52 14 PM" src="https://user-images.githubusercontent.com/68879427/109909108-5fbc2a00-7c5a-11eb-9b84-d8e5cdf99421.png">
